Sources/Client/Client/Client.swift

```diff
@@ -203,6 +203,21 @@ public final class Client {
             webSocket.disconnectInBackground()
         }
     }
+    
+    public func subscribe(forEvents eventTypes: Set<EventType> = Set(EventType.allCases), channelId: String? = nil, callback: @escaping OnEvent) -> Subscription {
+        let subscription = Subscription { [unowned self] uuid in
+            self.onEventObservers[uuid] = nil
+        }
+        
+        let handler: OnEvent = { event in
+            guard eventTypes.contains(event.type) && event.cid?.id == channelId else { return }
```

Comment:
```suggestion
            guard eventTypes.contains(event.type) else {
                return
            }
            
            if let cid = cid, event.cid != cid {
                return
            }
```

Reply:
Why do we need this? `event.cid == channelId` should be enough

Reply:
Because `cid` could be `nil` and `event.cid` could be nil.

Reply:
Yes, in that case we're handling it correctly.
```swift
func isChannelIdEqual(event: Event, channelId: ChannelId?) -> Bool {
    event.channelId == channelId
}

let channelIds = [ChannelId(type: .livestream, id: "gaming"), nil]

for firstChannelId in channelIds {
    for secondChannelId in channelIds {
        print("for \(firstChannelId) and \(secondChannelId) it's \(isChannelIdEqual(event: .init(channelId: firstChannelId), channelId: secondChannelId))")
    }
}
```
outputs:
```
for Optional(livestream:gaming) and Optional(livestream:gaming) it's true
for Optional(livestream:gaming) and nil it's false
for nil and Optional(livestream:gaming) it's false
for nil and nil it's true
```

Reply:
If you want to get all events for your solutions:
```swift
guard eventTypes.contains(event.type) && event.cid == cid else { return }
```
If parameter `cid` is nil, you'll miss all events where `event.cid` is nil.

Reply:
Now I get it 🤦‍♂ I got confused